src/main/src/test/java/org/geoserver/catalog/impl/CatalogImplTest.java

```diff
@@ -3514,6 +3514,165 @@ public void testFullTextSearchAddedKeyword() {
+        catalog.add(l2);
+
+        Filter filter = Predicates.exactTermSearch("temperature");
+        assertEquals(newHashSet(l, l2), asSet(catalog.list(LayerInfo.class, filter)));
+        assertEquals(newHashSet(ft, cv), asSet(catalog.list(ResourceInfo.class, filter)));
+        assertEquals(newHashSet(ft), asSet(catalog.list(FeatureTypeInfo.class, filter)));
+        assertEquals(newHashSet(cv), asSet(catalog.list(CoverageInfo.class, filter)));
+
+        filter = Predicates.exactTermSearch("air");
+        assertEquals(newHashSet(l), asSet(catalog.list(LayerInfo.class, filter)));
+        assertEquals(newHashSet(ft), asSet(catalog.list(ResourceInfo.class, filter)));
+        assertEquals(newHashSet(ft), asSet(catalog.list(FeatureTypeInfo.class, filter)));
+        assertTrue(asSet(catalog.list(CoverageInfo.class, filter)).isEmpty());
+
+        filter = Predicates.exactTermSearch("dewpointdepression");
+        assertEquals(newHashSet(l2), asSet(catalog.list(LayerInfo.class, filter)));
+        assertEquals(newHashSet(cv), asSet(catalog.list(ResourceInfo.class, filter)));
+        assertTrue(asSet(catalog.list(FeatureTypeInfo.class, filter)).isEmpty());
+        assertEquals(newHashSet(cv), asSet(catalog.list(CoverageInfo.class, filter)));
+
+        filter = Predicates.exactTermSearch("pressure");
+        assertTrue(asSet(catalog.list(LayerInfo.class, filter)).isEmpty());
+        assertTrue(asSet(catalog.list(ResourceInfo.class, filter)).isEmpty());
+        assertTrue(asSet(catalog.list(FeatureTypeInfo.class, filter)).isEmpty());
+        assertTrue(asSet(catalog.list(CoverageInfo.class, filter)).isEmpty());
+    }
+
+    @Test
+    public void testExactTermSearchAddedKeyword() {
+        ft.getKeywords().add(new Keyword("air_temp"));
+        ft.getKeywords().add(new Keyword("temperatureAir"));
+
+        l.setResource(ft);
+        addLayer();
+
+        LayerInfo lproxy = catalog.getLayer(l.getId());
+        FeatureTypeInfo ftproxy = (FeatureTypeInfo) lproxy.getResource();
+
+        ftproxy.getKeywords().add(new Keyword("newKeyword"));
+        catalog.save(ftproxy);
+
+        Filter filter = Predicates.exactTermSearch("newKeyword");
+        assertEquals(newHashSet(ftproxy), asSet(catalog.list(FeatureTypeInfo.class, filter)));
+        assertEquals(newHashSet(lproxy), asSet(catalog.list(LayerInfo.class, filter)));
+    }
+
+    @Test
+    public void testExclusiveExactTermSearch() {
+        ft.setName("water");
+        l.setResource(ft);
+
+        cv.setName("waterways");
+
+        addLayer();
+        catalog.add(cv);
+
+        LayerInfo l2 = newLayer(cv, s);
+        catalog.add(l2);
+
+        Filter filter = Predicates.exactTermSearch("water");
+        assertEquals(newHashSet(ft), asSet(catalog.list(ResourceInfo.class, filter)));
+        assertEquals(newHashSet(ft), asSet(catalog.list(FeatureTypeInfo.class, filter)));
+        assertTrue(asSet(catalog.list(CoverageInfo.class, filter)).isEmpty());
```

Comment:
In general, for new code, we try to avoid using assertTrue/assertFalse. Hamcrest provides a way to make the test fail with a more informative message, when that happens (assertTrue/assertFalse only report "I was expecting true but got false", Hamcrest reports "I was expecting an empty collection but got a, b, c")

```suggestion
        assertThat(asSet(catalog.list(CoverageInfo.class, filter)), Matchers.empty());
```

Reply:
Changed. Thanks for the feedback.